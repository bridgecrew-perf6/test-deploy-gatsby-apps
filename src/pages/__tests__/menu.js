import React from "react"
import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"
import Menu from "../menu"

const menuData = {
  allMenuDataJson: {
    edges: [
      {
        node: {
          drinks: {
            heading: "Drinks",
            text: "Our Drink Menu",
            allDrinks: [
              {
                name: "Drip Coffee",
                price: "2.30",
              },
            ],
          },
        },
      },
      {
        node: {
          eats: {
            heading: "Eats",
            text: "Our Food Menu",
            allEats: [
              {
                name: "Breakfast Sandwich",
                price: "5.30",
              },
            ],
          },
        },
      },
    ],
  },
}

describe("header", () => {
  // it("renders Mukti logo image", () => {
  //   render(<Menu data={menuData} />)
  //   const logo = screen.getByAltText("logo")
  //   expect(logo).toBeInTheDocument()
  // })

  it("renders cart image", () => {
    render(<Menu data={menuData} />)
    const logo = screen.getByAltText("cart")
    expect(logo).toBeInTheDocument()
  })
})

describe("renders menu cards", () => {
  it("renders Menu correctly", () => {
    const tree = renderer.create(<Menu data={menuData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders the menu text correctly", () => {
    render(<Menu data={menuData} />)
    const eatMenuItem = screen.getByText("Breakfast Sandwich")
    const eatItemPrice = screen.getByText("5.30")
    const drinkHeading = screen.getByText("Drinks")
    const drinkText = screen.getByText("Our Drink Menu")
    expect(eatMenuItem).toBeInTheDocument()
    expect(eatItemPrice).toBeInTheDocument()
    expect(drinkHeading).toBeInTheDocument()
    expect(drinkText).toBeInTheDocument()
  })

  it("renders correct number of buttons", () => {
    render(<Menu data={menuData} />)
    expect(screen.getAllByRole("button")).toHaveLength(6)
  })

  it("renders correct number add icons menu items", () => {
    render(<Menu data={menuData} />)
    expect(screen.getAllByAltText("add-item")).toHaveLength(2)
  })
})
