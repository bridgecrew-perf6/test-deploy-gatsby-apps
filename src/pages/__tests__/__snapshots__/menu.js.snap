// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders menu cards renders Menu correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      style={
        Object {
          "flexGrow": "1",
        }
      }
    >
      <header
        className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
        style={
          Object {
            "boxShadow": "0px 0px 0px 0px",
          }
        }
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-17"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <img
                alt="logo"
                className="makeStyles-logo-18"
                src="test-file-stub"
              />
            </span>
          </button>
          <div
            className="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-13 MuiToolbar-gutters"
          >
            <a
              className="makeStyles-route-14"
              href="/menu"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-routeButton-15"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  DRINKS & FOOD
                </span>
              </button>
            </a>
            <a
              className="makeStyles-route-14"
              href="/about"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-routeButton-15"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  ABOUT
                </span>
              </button>
            </a>
            <a
              className="makeStyles-route-14"
              href="/cart"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-routeButton-15"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <img
                    alt="cart"
                    className="makeStyles-cartImg-16"
                    src="test-file-stub"
                  />
                  <p
                    className="MuiTypography-root MuiTypography-body1"
                  >
                    
                  </p>
                </span>
              </button>
            </a>
          </div>
        </div>
      </header>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    style={
      Object {
        "paddingTop": "20vh",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <main>
          <div
            style={
              Object {
                "padding": "40px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
            >
              <h3
                className="MuiTypography-root MuiTypography-h3"
              >
                Drinks
              </h3>
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
              >
                Our Drink Menu
              </h6>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-9"
              style={
                Object {
                  "marginTop": "10vh",
                }
              }
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
              >
                <div
                  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
                  style={
                    Object {
                      "padding": "10px",
                    }
                  }
                >
                  <div
                    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-direction-xs-column MuiGrid-justify-xs-space-between"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-item"
                    >
                      <h6
                        className="MuiTypography-root MuiTypography-subtitle2"
                      >
                        Drip Coffee
                      </h6>
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
                      >
                        <div
                          className="MuiGrid-root MuiGrid-item"
                        >
                          <p
                            className="MuiTypography-root MuiTypography-body2"
                          >
                            2.30
                          </p>
                        </div>
                        <div
                          className="MuiGrid-root MuiGrid-item"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-19 MuiIconButton-sizeSmall"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <span
                              className="MuiIconButton-label"
                            >
                              <img
                                alt="add-item"
                                src="test-file-stub"
                                style={
                                  Object {
                                    "height": "6vh",
                                    "width": "6vw",
                                  }
                                }
                              />
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "padding": "40px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
            >
              <h3
                className="MuiTypography-root MuiTypography-h3"
              >
                Eats
              </h3>
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
              >
                Our Food Menu
              </h6>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-9"
              style={
                Object {
                  "marginTop": "10vh",
                }
              }
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
              >
                <div
                  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
                  style={
                    Object {
                      "padding": "10px",
                    }
                  }
                >
                  <div
                    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-direction-xs-column MuiGrid-justify-xs-space-between"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-item"
                    >
                      <h6
                        className="MuiTypography-root MuiTypography-subtitle2"
                      >
                        Breakfast Sandwich
                      </h6>
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
                      >
                        <div
                          className="MuiGrid-root MuiGrid-item"
                        >
                          <p
                            className="MuiTypography-root MuiTypography-body2"
                          >
                            5.30
                          </p>
                        </div>
                        <div
                          className="MuiGrid-root MuiGrid-item"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-19 MuiIconButton-sizeSmall"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <span
                              className="MuiIconButton-label"
                            >
                              <img
                                alt="add-item"
                                src="test-file-stub"
                                style={
                                  Object {
                                    "height": "6vh",
                                    "width": "6vw",
                                  }
                                }
                              />
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-footerContainer-11 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="logo"
          className="makeStyles-footerLogo-20"
          src="test-file-stub"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          2835 Cook Street Village
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
        >
          Victoria, British Columbia
        </p>
      </div>
    </div>
  </div>
</div>
`;
